subsys/bluetooth/controller/ll_sw/ull_master.c

```diff
@@ -392,22 +392,47 @@ uint8_t ll_connect_enable(uint8_t is_coded_included)
 
 uint8_t ll_connect_disable(void **rx)
 {
+	struct ll_scan_set *scan_coded;
+	struct lll_scan *scan_lll;
 	struct lll_conn *conn_lll;
 	struct ll_scan_set *scan;
-	uint8_t status;
+	uint8_t err;
+
+	scan = ull_scan_is_enabled_get(SCAN_HANDLE_1M);
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT) &&
+	    IS_ENABLED(CONFIG_BT_CTLR_PHY_CODED)) {
+		scan_coded = ull_scan_is_enabled_get(SCAN_HANDLE_PHY_CODED);
+	} else {
+		scan_coded = NULL;
+	}
 
-	scan = ull_scan_is_enabled_get(0);
 	if (!scan) {
-		return BT_HCI_ERR_CMD_DISALLOWED;
+		if (!scan_coded) {
+			return BT_HCI_ERR_CMD_DISALLOWED;
+		}
+
+		scan_lll = &scan_coded->lll;
+	} else {
+		scan_lll = &scan->lll;
 	}
 
-	conn_lll = scan->lll.conn;
+	conn_lll = scan_lll->conn;
 	if (!conn_lll) {
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
-	status = ull_scan_disable(0, scan);
-	if (!status) {
+	if (scan) {
+		err = ull_scan_disable(SCAN_HANDLE_1M, scan);
+	} else {
+		err = 0U;
+	}
+
+	if (!err && scan_coded) {
```

Comment:
Shouldn't this try to disable the Coded PHY scanner even if there was an error with disabling the 1M PHY scanner?

Reply:
Until this line, with `ull_scan_disable` if fails, no internal context has been modified. And, as `scan` is non-NULL because it shall be enabled, a failure here indicates a flaw, hence, exiting without further modification to the context of the controller.